src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml

```diff
@@ -76,212 +125,233 @@ limitations under the License.
+    <item name="HeatingWithReheat" value="0x03"/>
+    <item name="CoolingAndHeating" value="0x04"/>
+    <item name="CoolingAndHeatingWithReheat" value="0x05"/>
   </enum>
 
-  <enum name="TemperatureSetpointHold" type="enum8">
+  <enum name="TemperatureSetpointHoldEnum" type="enum8">
+    <cluster code="0x0201"/>
     <item name="SetpointHoldOff" value="0x00"/>
     <item name="SetpointHoldOn" value="0x01"/>
   </enum>
 
-  <enum name="SetpointAdjustMode" type="enum8">
+  <enum name="SetpointRaiseLowerModeEnum" type="enum8">
     <cluster code="0x0201"/>
-    <item name="Heat" value="0x0"/>
-    <item name="Cool" value="0x1"/>
-    <item name="Both" value="0x2"/>
+    <item name="Heat" value="0x00"/>
+    <item name="Cool" value="0x01"/>
+    <item name="Both" value="0x02"/>
+  </enum>
+  
+  <enum name="ACCapacityFormatEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="BTUh" value="0x00"/>
+  </enum>
+
+  <enum name="ACCompressorTypeEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Unknown" value="0x00"/>
+    <item name="T1" value="0x01"/>
+    <item name="T2" value="0x02"/>
+    <item name="T3" value="0x03"/>
+  </enum>
+
+  <enum name="ACLouverPositionEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Closed" value="0x01"/>
+    <item name="Open" value="0x02"/>
+    <item name="Quarter" value="0x03"/>
+    <item name="Half" value="0x04"/>
+    <item name="ThreeQuarters" value="0x05"/>
+  </enum>
+
+  <enum name="ACRefrigerantTypeEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Unknown" value="0x00"/>
+    <item name="R22" value="0x01"/>
+    <item name="R410a" value="0x02"/>
+    <item name="R407c" value="0x03"/>
+  </enum>
+
+  <enum name="ACTypeEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Unknown" value="0x00"/>
+    <item name="CoolingFixed" value="0x01"/>
+    <item name="HeatPumpFixed" value="0x02"/>
+    <item name="CoolingInverter" value="0x03"/>
+    <item name="HeatPumpInverter" value="0x04"/>
+  </enum>
+
+  <enum name="SetpointChangeSourceEnum" type="enum8">
+    <cluster code="0x0201"/>
+    <item name="Manual" value="0x00"/>
+    <item name="Schedule" value="0x01"/>
+    <item name="External" value="0x02"/>
   </enum>
 
-  <struct name="ThermostatScheduleTransition">
+  <struct name="WeeklyScheduleTransitionStruct">
     <cluster code="0x0201"/>
-    <item fieldId="0" name="TransitionTime" type="int16u"/>
     <!-- See https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6217 for HeatSetpoint and CoolSetpoint.  They might end up being nullable. -->
-    <item fieldId="1" name="HeatSetpoint" type="int16s" isNullable="true"/>
-    <item fieldId="2" name="CoolSetpoint" type="int16s" isNullable="true"/>
+    <item fieldId="0" name="TransitionTime" type="int16u" min="0" max="1439"/>
+    <item fieldId="1" name="HeatSetpoint" type="temperature"/>
+    <item fieldId="2" name="CoolSetpoint" type="temperature"/>
```

Comment:
This change from nullable to not nullable is a wire format compat break.  We can't make that change.  The spec needs to change here.

Reply:
@hasty